src/assemble_workflow/dist.py

```diff
@@ -36,42 +41,145 @@ def extract(self, dest):
 
         raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), os.path.join(dest, "*"))
 
-    def build(self, name, dest):
-        self.__build__(name, dest)
+    def build(self, bundle_recorder, dest):
+        name = bundle_recorder.package_name
+        self.__build__(bundle_recorder, dest)
         path = os.path.join(dest, name)
         shutil.copyfile(name, path)
         logging.info(f"Published {path}.")
 
     @classmethod
-    def from_path(cls, name, path):
-        ext = os.path.splitext(path)[1]
-        if ext == ".gz":
-            return DistTar(name, path)
-        elif ext == ".zip":
-            return DistZip(name, path)
+    def create_dist(cls, name, path, distribution):
+        DISTRIBUTIONS = {
```

Comment:
Move the constant out of the method and make a map between the extension and the class name, so:

```python
"rpm":DistRpm
...
```

then `return DISTRIBUTIONS[distribution](name, path)`, avoids having to copy constructor args and keeps the table a constant.

Reply:
In this way, the class name will not be defined. How about keep the dictionary in the method but remove all the `(name, path)` ?

Reply:
The pattern we use is to use another file called `Dists` and have the method there. Otherwise you're introducing a circular dependency.